dll/ntdll/rtl/libsupp.c

```diff
@@ -505,6 +505,62 @@ RtlpGetAtomEntry(PRTL_ATOM_TABLE AtomTable, ULONG Index)
    return NULL;
 }
 
+/* Ldr SEH-Protected access to IMAGE_NT_HEADERS */
+
+/* Rtl SEH-Free version of this */
+NTSTATUS
+NTAPI
+RtlpImageNtHeaderEx(
+    _In_ ULONG Flags,
+    _In_ PVOID Base,
+    _In_ ULONG64 Size,
+    _Out_ PIMAGE_NT_HEADERS *OutHeaders);
+
+
+/*
+ * @implemented
+ * @note: This is here, so that we do not drag SEH into rosload, freeldr and bootmgfw
+ */
+NTSTATUS
+NTAPI
+RtlImageNtHeaderEx(
+    _In_ ULONG Flags,
+    _In_ PVOID Base,
+    _In_ ULONG64 Size,
+    _Out_ PIMAGE_NT_HEADERS *OutHeaders)
+{
+    NTSTATUS Status;
+
+    _SEH2_TRY
+    {
+        Status = RtlpImageNtHeaderEx(Flags, Base, Size, OutHeaders);
```

Comment:
Since `RtlpImageNtHeaderEx` sets `*OutHeaders = NULL` as its first operation, I guess we're fine wrt making sure that always happens in case of an exception.
We'll have slightly different behavior here in the case where `OutHeaders` is an invalid but non-null pointer (we'll catch the exception and return, whereas the Windows implementation probably just crashes). Could be worth doing an explicit `*OutHeaders = NULL` in the exception case, or could be useless code that's never gonna be relevant :p

Reply:
This is a cornercase, but also something that might be very hard to find if it causes problems.
How about I set the OutHeaders to `NULL` before entering SEH?

Reply:
Yeah that would work too. Just need to handle the `OutHeaders == NULL` case, too, I suppose.